.github/workflows/push.yml

```
@@ -17,6 +17,9 @@ on:
   pull_request:
     branches:
       - '**'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.url || github.run_id }}
+  cancel-in-progress: true
```

```suggestion
  group: ${{ github.workflow }}-${{ github.event.pull_request.url }}
  cancel-in-progress: ${{ github.event.pull_request.url }}
```

it should apply to PRs only, not pushes to master/release branches, nor pushes to forks

---

as consequence `|| github.run_id` is different for each push?

if my sudgestion works, I would say it is much more readable

---

Look at the docs, that's not how it's described, although the docs also aren't great.

---

No, it will not apply to stable branches, `cancel-in-progress: ${{ github.event.pull_request.url }}` will make sure such limitation will apply only to PRs.

---

Please read the docs. This adds behavior even with `cancel-in-progress: false`.